core/src/test/java/org/apache/iceberg/rest/responses/TestLoadTableResponse.java

```diff
@@ -0,0 +1,218 @@
+            .withMetadataLocation(TEST_METADATA_LOCATION)
+            .build();
+
+    return LoadTableResponse.builder()
+        .withTableMetadata(metadata)
+        .addAllConfig(CONFIG)
+        .build();
+  }
+
+  @Override
+  public LoadTableResponse deserialize(String json) throws JsonProcessingException {
+    LoadTableResponse resp = mapper().readValue(json, LoadTableResponse.class);
+    resp.validate();
+    return resp;
+  }
+
+  @Test
+  public void testFailures() {
+    AssertHelpers.assertThrows(
+        "Table metadata should be required",
+        NullPointerException.class,
+        "Invalid metadata: null",
+        () -> LoadTableResponse.builder().build());
+  }
+
+  @Test
+  public void testRoundTripSerde() throws Exception {
+    // Default fields are missing in this JSON
+    String tableMetadataV1Json = readTableMetadataInputFile("TableMetadataV1Valid.json");
+    TableMetadata metadataV1 = TableMetadataParser.fromJson(null, TEST_METADATA_LOCATION, tableMetadataV1Json);
+    // Convert the TableMetadata JSON from the file to an object and then back to JSON so that missing fields
+    // are filled in with their default values.
+    String json1 = String.format(
+        "{\"metadata-location\":\"%s\",\"metadata\":%s,\"config\":{\"foo\":\"bar\"}}",
+        TEST_METADATA_LOCATION, TableMetadataParser.toJson(metadataV1));
+    LoadTableResponse resp1 = LoadTableResponse.builder()
+        .withTableMetadata(metadataV1)
+        .addAllConfig(CONFIG)
+        .build();
+    assertRoundTripSerializesEquallyFrom(json1, resp1);
+
+    String tableMetadataV2Json = readTableMetadataInputFile("TableMetadataV2Valid.json");
+    TableMetadata metadataV2 = TableMetadataParser.fromJson(null, TEST_METADATA_LOCATION, tableMetadataV2Json);
+    // Convert the TableMetadata JSON from the file to an object and then back to JSON so that missing fields
+    // are filled in with their default values.
+    String json2 = String.format(
+        "{\"metadata-location\":\"%s\",\"metadata\":%s,\"config\":{\"foo\":\"bar\"}}",
+        TEST_METADATA_LOCATION, TableMetadataParser.toJson(metadataV2));
+    LoadTableResponse resp2 = LoadTableResponse.builder()
+        .withTableMetadata(metadataV2)
+        .addAllConfig(CONFIG)
+        .build();
+    assertRoundTripSerializesEquallyFrom(json2, resp2);
+  }
+
+  @Test
+  public void testCanDeserializeWithoutDefaultValues() throws Exception {
+    String metadataJson = readTableMetadataInputFile("TableMetadataV1Valid.json");
+    String json1 = String.format(
+        "{\"metadata-location\":\"%s\",\"metadata\":%s,\"config\":{\"foo\":\"bar\"}}",
+        TEST_METADATA_LOCATION, metadataJson);
+
+    TableMetadata metadataV1 = TableMetadataParser.fromJson(null, TEST_METADATA_LOCATION, metadataJson);
+    LoadTableResponse resp1 = LoadTableResponse.builder()
+        .withTableMetadata(metadataV1)
+        .addAllConfig(CONFIG)
+        .build();
+    assertEquals(deserialize(json1), resp1);
```

Comment:
This is the first half of what `assertRoundTripSerializesEquallyFrom` will check and the case is identical to the one above in `testRoundTripSerde`.

From the test name, I think what you intended to test is that all values that can be omitted will be deserialized to defaults. Should this omit "config" from the JSON string and validate that the response is equal to the one created without calling `addAllConfig`?

Also, there's no need for `json1` and `resp1` names since this only has one JSON string and one response object.

Reply:
> From the test name, I think what you intended to test is that all values that can be omitted will be deserialized to defaults. Should this omit "config" from the JSON string and validate that the response is equal to the one created without calling addAllConfig?

Yeah that's what this is testing. It's a little unclear from the way it's written admittedly.

I'll try removing `config` and then ensuring we get the default back.

And update the names.

Reply:
> Yeah that's what this is testing. It's a little unclear from the way it's written admittedly.

I'm not sure I understand. What was this testing in the version I was looking at?

Initially, I was going to comment that it duplicates the first part of the round-trip test for v1. Then I realized that it was intended to test something else.

Reply:
In the version you were looking at, this was testing that the JSON parses into the class as expected, even if default values are not present.

Because default values are not present in the external JSON, I used those to represent missing values (which was the source of confusion).

For the `round trip` tests, it tests that the JSON deserializes into the object _as well as that the JSON matches exactly as expected the serialiazed output of the object_

That's the `round trip` in the `assertRoundTripSerialzesEquallyFrom`: https://github.com/apache/iceberg/blob/22298818cb7eba13294eda3b2dd696b773e0f170/core/src/test/java/org/apache/iceberg/rest/RequestResponseTestBase.java#L102-L109

Reply:
Do let me know if you think it would be better some other way (for this test suite at least). Though I'm anxious to get it in.